src/ppl/src/Slots.cpp

```diff
@@ -58,4 +58,10 @@ int Section::getMaxContiguousSlots(const std::vector<Slot>& slots)
   return max_contiguous_slots;
 }
 
+bool Interval::operator==(const Interval& interval) const
+{
+  return edge_ == interval.getEdge() && begin_ == interval.getBegin()
+         && end_ == interval.getEnd() && layer_ == interval.getLayer();
+}
+
```

Comment:
This looks like it is equivalent to what the compiler would do automatically.  What makes this required?

Reply:
`Interval` is a class.  When I use the `==`operator directly, the build fails with
```
IOPlacer.cpp:1421:29: error: invalid operands to binary expression ('ppl::Interval' and 'ppl::Interval')

    if (constraint.interval == interval) {
```

Reply:
Nvm, I forgot operator== isn't one that is default generated (I always think it should be).